virttest/test_setup.py

```diff
@@ -434,14 +434,15 @@ def __init__(self, params):
         vm_mem_minimum = params.get("vm_mem_minimum", "512M")
         self.lowest_mem_per_vm = float(normalize_data_size(vm_mem_minimum))
         utils_memory.drop_caches()
+        self.hugepages_surp = self.get_hugepages_surp()
 
         target_hugepages = params.get("target_hugepages")
         if target_hugepages is None:
             target_hugepages = self.get_target_hugepages()
         else:
             target_hugepages = int(target_hugepages)
 
-        self.target_hugepages = target_hugepages
+        self.target_hugepages = target_hugepages + self.hugepages_surp
```

Comment:
why do we include `hugepage_surp` with target_hugepages ?
if we don't really test conditions for surplus hugepages then ideally we have to set `nr_overcommit_hugepages` to zero so that kernel doesn't try to allocate hugepages from normal page pool.

Reply:
target_hugepages will calculate the huge page it wants, if hugepage_surp is not zero then the guest will not be launched.
Set `nr_overcommit_hugepages` to zero is a better method, I will update with your suggestion.

Reply:
Thank you.

Reply:
one more thing, if you really have to test hugepages_surp, then we should allocate target_hugepages less than what guest required so that kernel will be forced to allocate hugepages from normal pages pool to meet VM to be launched. I think if we allocate `target_hugepages + self.hugepages` then we really don't exercise `hugepage_surp`. please share your thoughts.

Reply:
I just have a test case that has failed because of hugepages_surp and I don't test it now.
If we want to let target_hugepages less than what guest required, we can force set `target_hugepages = num` on our test environment.
`target_hugepages + self.hugepages` really not a good way.